src/psyclone/psyir/backend/sir.py

```diff
@@ -520,3 +552,39 @@ def schedule_node(self, node):
         for child in node.children:
             result += self._visit(child)
         return result
+
+    def naryoperation_node(self, node):
+        '''This method is called when an NaryOperation instance is found in
+        the PSyIR tree.
+
+        :param node: a NaryOperation PSyIR node.
+        :type node: :py:class:`psyclone.psyir.nodes.NaryOperation`
+
+        :returns: the SIR Python code.
+        :rtype: str
+
+        :raises VisitorError: if there is no mapping from the PSyIR \
+        operator to SIR.
+
+        '''
+        nary_operators = {
```

Comment:
Please add a comment giving the purpose of this dict - I guess this relates to the comment at L587?

Reply:
Added.